src/components/Editor/BacklinkSuggestionsDisplay.tsx

```diff
@@ -0,0 +1,85 @@
+import { removeFileExtension } from "@/functions/strings";
+import React, { useRef, useEffect, useState, useMemo } from "react";
+
+export interface SuggestionsState {
+  textWithinBrackets: string;
+  position: { top: number; left: number };
+  onSelect: (suggestion: string) => void;
+}
+
+interface SuggestionsDisplayProps {
+  suggestionsState: SuggestionsState;
+  suggestions: string[];
+}
+
+const InEditorBacklinkSuggestionsDisplay: React.FC<SuggestionsDisplayProps> = ({
+  suggestionsState,
+  suggestions,
+}) => {
+  const suggestionsRef = useRef<HTMLDivElement | null>(null);
+  const [layout, setLayout] = useState({
+    top: -9999,
+    left: -9999,
+    display: "none",
+  });
+
+  const filteredSuggestions = useMemo(() => {
+    if (!suggestionsState.textWithinBrackets) return [];
+    console.log("suggestionsState.text", suggestionsState.textWithinBrackets);
+    const lowerCaseText = suggestionsState.textWithinBrackets.toLowerCase();
+    return suggestions
+      .filter((suggestion) => suggestion.toLowerCase().includes(lowerCaseText))
+      .map(removeFileExtension)
+      .slice(0, 5);
+  }, [suggestions, suggestionsState.textWithinBrackets]);
+
+  useEffect(() => {
+    if (
+      !suggestionsState.position ||
+      filteredSuggestions.length === 0 ||
+      !suggestionsRef.current
+    ) {
+      return;
+    }
+    const { top, left } = suggestionsState.position;
```

Comment:
Minor usability suggestion - we should have the bounding box for displays just outside of the square brackets, rather right next to the cursor. That makes it easier to edit 

![image](https://github.com/reorproject/reor/assets/58682208/5b87900b-69a4-4803-aca4-73062e64f65b)

Reply:
done